### ISSUE TITLE
underscores get propagated into invalid names

### ISSUE THREAD

**Issue by idcrook** [kubernetes-incubator/kompose#420]
searched for dupes, and found none.

docker compose file has name like: `phant_server:` in yaml, which contain embedded underscores. `kompose` does not normalize the name, resulting in invalid names being used.

#### pvc

```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  creationTimestamp: null
  name: phant_server-claim0
...
```

yields error

```
The PersistentVolume "phant_server-persistent-volume" is invalid: metadata.name: Invalid value: "phant_server-persistent-volume": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)* (e.g. 'example.com')
```

#### deployment 

```
The Deployment "phant_server" is invalid:
* metadata.name: Invalid value: "phant_server": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)* (e.g. 'example.com')
```

## possible normalization suggestions

phant_server -> phant.server
phant_server -> phant-server
phant_server -> phantserver

**Comment by surajssd** [comment]
@idcrook see if it works for you!

**Comment by idcrook** [comment]
Thx

**Comment by kadel** [comment]
Opening this issue again as #429 fixed this problem only for service names, same problem is still in volume names.

**Comment by surajssd** [comment]
@kadel so @containscafeine found that there is a caveat with what #429 did.

Let say if a user defines a docker-compose file as

```bash
services:
  foo-bar:
...
  blah-blah:
...

``` 


Now if the `blah-blah` service tries to access `foo-bar` it will simply call out on `foo-bar` but here we are converting `foo-bar` to `foo_bar` so now the name resolution works in docker-compose but not in k8s because we just changed the name in config but code is using old name.

So we need to give out a WARNING saying this is done.

**Comment by surajssd** [comment]
so above is filed in https://github.com/kubernetes-incubator/kompose/issues/433

**Comment by procrypt** [comment]
@surajssd @kadel Taking this up.

**Comment by surajssd** [comment]
@procrypt i think @surajnarwade is working on this

**Comment by surajnarwade** [comment]
@surajssd @procrypt , its okay, I havn't started yet, will pick other one :)

**Comment by kadel** [comment]
Ah, sorry @surajnarwade @procrypt I didn't know that, or even more probably forgot about it :blush:

**Comment by idcrook** [comment]
Can be closed when #509 is released

**Comment by xiaoping378** [comment]
```
version: '2'

services:

  ca.org1.example.com:
```
docker-compose like upon, when "kubectl create" will ouput error 
```
The Service "orderer.example.com" is invalid: metadata.name: Invalid value: "orderer.example.com": a DNS-1035 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')

```
